bentoml/service/env.py

```diff
@@ -323,29 +322,42 @@ def save(self, path):
         # Parse pip packages from specified requirements.txt file, note that this file
         # may not be presented in the docker container, but its content will be parsed
         # and added to the new requirements.txt file generated in the Bento
-        if self._requirements_txt_file:
-            self.add_packages_from_requirements_txt_file(self._requirements_txt_file)
 
         conda_yml_file = os.path.join(path, "environment.yml")
         self._conda_env.write_to_yaml_file(conda_yml_file)
 
         with open(os.path.join(path, "python_version"), "wb") as f:
             f.write(self._python_version.encode("utf-8"))
 
+        # If requirements.txt is specified, other requirements will
+        # NOT be parsed (#1421)
         requirements_txt_file = os.path.join(path, "requirements.txt")
-        with open(requirements_txt_file, "wb") as f:
-            if self._pip_index_url:
-                f.write(f"--index-url={self._pip_index_url}\n".encode("utf-8"))
-            if self._pip_trusted_host:
-                f.write(f"--trusted-host={self._pip_trusted_host}\n".encode("utf-8"))
-            if self._pip_extra_index_url:
-                f.write(
-                    f"--extra-index-url={self._pip_extra_index_url}\n".encode("utf-8")
-                )
-            pip_content = '\n'.join(
-                [str(pkg_req) for pkg_req in self._pip_packages.values()]
-            ).encode("utf-8")
-            f.write(pip_content)
+        if self._requirements_txt_file:
+            logger.info(
```

Comment:
remove this logging here and show a warning in the `@env` constructor instead? that way we can give the user more specific warning and help message

Reply:
in `env_decorator`, we can have something like this:

```python
if requirements_txt_file:
    if pip_packages:
         logger.warn(...)
    if infer_pip_packages is not None:
         logger.warn(...)
    if pip_index_url or pip_trusted_host or pip_extra_index_url:
         # tell user to put those options in the requirements txt file directly instead
         logger.warn(...)
```

Reply:
why would we want to warn on `infer_pip_packages is not None`?

Reply:
@jackyzha0  because `infer_pip_packages` controls if BentoML tries to add pip packages to the `self.packages` list, which will be ignored if `requirements_txt_file` is provided